doc/doc-support/generate-function-docs.nix

```diff
@@ -1,27 +1,31 @@
 # Generates the documentation for library functions via nixdoc.
 
-{ pkgs, nixpkgs, libsets }:
+{ pkgs, nixpkgs, libsets, library, src, name, prefix }:
 
 with pkgs;
 
 let
-  locationsJSON = import ./lib-function-locations.nix { inherit pkgs nixpkgs libsets; };
+  locationsJSON = import ./lib-function-locations.nix { inherit pkgs nixpkgs libsets library prefix; };
 in
 stdenv.mkDerivation {
-  name = "nixpkgs-lib-docs";
-  src = ../../lib;
+  inherit src name;
 
   buildInputs = [ nixdoc ];
+  env = {
+    # lower case $prefix is written by setup.sh for ./configure when empty
+    PREFIX = prefix;
+    PREFIXDOT = if prefix == "" then "" else "${prefix}.";
+  };
   installPhase = ''
     function docgen {
       name=$1
       baseName=$2
       description=$3
-      # TODO: wrap lib.$name in <literal>, make nixdoc not escape it
+      # TODO: wrap ${prefix}$name in <literal>, make nixdoc not escape it
       if [[ -e "../lib/$baseName.nix" ]]; then
-        nixdoc -c "$name" -d "lib.$name: $description" -l ${locationsJSON} -f "$baseName.nix" > "$out/$name.md"
+        nixdoc -c "$name" -d "$PREFIXDOT$name: $description" -l ${locationsJSON} -f "$baseName.nix" --prefix "$PREFIX" > "$out/$name.md"
```

Comment:
This would require nixdoc to support that feature as you already mentioned. I could take care of that in the next couple of days.